core/iwasm/common/wasm_runtime_common.c

```diff
@@ -181,15 +181,40 @@ static RunningMode runtime_running_mode = Mode_Default;
    of signal handler */
 static os_thread_local_attribute WASMExecEnv *exec_env_tls = NULL;
 
+static bool
+is_sig_addr_oob(void *sig_addr, WASMModuleInstance *module_inst)
+{
+    WASMMemoryInstance *memory_inst;
+    uint8 *mapped_mem_start_addr = NULL;
+    uint8 *mapped_mem_end_addr = NULL;
+    uint32 i;
+
+    for (i = 0; i < module_inst->memory_count; ++i) {
+        /* To be compatible with multi memory, get the ith memory instance */
+        memory_inst = wasm_get_memory_i(module_inst, i);
+        if (memory_inst) {
```

Comment:
when can this be NULL here?

Reply:
I am not sure, I see that the original code handles that condition so I just leave it there. I will double check whether it's necessary

Reply:
the original code needed to handle the "no memory" case. this loop doesn't need it.